src/test/kotlin/com/terraformation/backend/customer/OrganizationServiceTest.kt

```diff
@@ -232,4 +237,57 @@ internal class OrganizationServiceTest : DatabaseTest(), RunsAsUser {
 
     publisher.assertEventPublished(OrganizationDeletionStartedEvent(organizationId))
   }
+
+  @Test
+  fun `UserAddedToOrganization event is published when existing user is added to an organization`() {
+    val otherUserId = UserId(100)
+    val organizationId = OrganizationId(1)
+
+    insertUser(user.userId)
+    insertUser(otherUserId, email = "existingUser@email.com")
+
+    insertOrganization(organizationId)
+
+    every { user.canAddOrganizationUser(organizationId) } returns true
+    every { user.canSetOrganizationUserRole(organizationId, Role.Contributor) } returns true
+    service.addUser(
+        email = "existingUser@email.com", organizationId = organizationId, role = Role.Contributor)
+
+    publisher.assertExactEventsPublished(
+        setOf(
+            UserAddedToOrganizationEvent(
+                userId = otherUserId, organizationId = organizationId, addedBy = user.userId)))
+  }
+
+  @Test
+  fun `UserAddedToTerraware event is published when new user is added to an organization`() {
+    val otherUserEmail = "newuser@email.com"
+    val otherUserId = UserId(100)
+    val organizationId = OrganizationId(1)
+
+    insertUser(user.userId)
+    insertUser(userId = otherUserId, email = otherUserEmail)
+
+    val otherUser = userStore.fetchByEmail(otherUserEmail)
+    assertNotNull(otherUser, "No new user")
+
+    insertOrganization(organizationId)
+
+    every { user.canAddOrganizationUser(organizationId) } returns true
+    every { user.canSetOrganizationUserRole(organizationId, Role.Contributor) } returns true
+    // mock missing user on initial check
+    every { userStore.fetchByEmail(otherUserEmail) } returnsMany listOf(null, otherUser)
```

Comment:
I started out that way but was running into db errors (insert was failing on duplicate users_pkey, which I thought was strange - since that's a db generated id). Will revisit this, more alert now than when I last worked on this test.